tests/unit/s2n_key_material_test.c

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "utils/s2n_key_material.h"
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_cipher_suites.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_crypto.h"
+
+S2N_RESULT helper_generate_test_data(struct s2n_blob *test_data)
+{
+    struct s2n_stuffer test_data_stuffer = { 0 };
+    EXPECT_SUCCESS(s2n_stuffer_init(&test_data_stuffer, test_data));
```

Comment:
If you're in a function that returns S2N_RESULT, you should probably call RESULT_GUARD_POSIX instead. EXPECT_SUCCESS will exit if it fails, which can be confusing if this method is called in multiple places.

Reply:
This wasn't resolved I think?

Reply:
yep seems like i missed a few while juggling changes. Let me look thru all of them to make sure that I addressed them all.

Reply:
addressed in https://github.com/aws/s2n-tls/pull/4047/commits/39a166ee4dd0facfe8d6d3111d00773fa18100b5